src/analysis/retail/monk/windwalker/modules/apl/AplCheck.tsx

```diff
@@ -1,63 +1,110 @@
-import aplCheck, { build, Condition } from 'parser/shared/metrics/apl';
+import aplCheck, { build, PlayerInfo } from 'parser/shared/metrics/apl';
 import {
   and,
   buffMissing,
   buffPresent,
   describe,
   hasResource,
+  hasTalent,
   lastSpellCast,
   not,
-  or,
 } from 'parser/shared/metrics/apl/conditions';
 import SPELLS from 'common/SPELLS';
 import { suggestion } from 'parser/core/Analyzer';
 import annotateTimeline from 'parser/shared/metrics/apl/annotate';
 import TALENTS from 'common/TALENTS/monk';
 import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
 import { SpellLink } from 'interface';
+import { AnyEvent } from 'parser/core/Events';
+import { serenityDurationRemainingLT } from 'analysis/retail/monk/windwalker/modules/apl/serenityDurationRemaining';
+import { AplRuleProps } from 'parser/shared/metrics/apl/ChecklistRule';
 
-const serenityOr = (cond: Condition<any>) => or(cond, buffPresent(TALENTS.SERENITY_TALENT));
-
+const inSerenity = buffPresent(TALENTS.SERENITY_TALENT);
 const hasChi = (min: number) => hasResource(RESOURCE_TYPES.CHI, { atLeast: min });
 
-export const apl = build([
+export const serenityApl = build([
+  {
+    spell: TALENTS.FISTS_OF_FURY_TALENT,
+    condition: and(inSerenity, serenityDurationRemainingLT(1500)),
+  },
+  {
+    spell: TALENTS.FAELINE_STOMP_TALENT,
+    condition: and(
+      hasTalent(TALENTS.FAELINE_HARMONY_TALENT),
+      buffMissing(SPELLS.FAELINE_HARMONY_BUFF),
+    ),
+  },
+  {
+    spell: TALENTS.STRIKE_OF_THE_WINDLORD_TALENT,
+    condition: inSerenity,
+  },
+  {
+    spell: TALENTS.FISTS_OF_FURY_TALENT,
+    condition: inSerenity,
+  },
+  {
+    spell: TALENTS.RISING_SUN_KICK_TALENT,
+    condition: inSerenity,
+  },
+  {
+    spell: SPELLS.BLACKOUT_KICK,
+    condition: inSerenity,
+  },
+  {
+    spell: SPELLS.SPINNING_CRANE_KICK,
+    condition: and(inSerenity, buffPresent(TALENTS.DANCE_OF_CHI_JI_TALENT)),
+  },
+  {
+    spell: TALENTS.CHI_WAVE_TALENT,
+    condition: inSerenity,
+  },
+  {
+    spell: SPELLS.TIGER_PALM,
+    condition: inSerenity,
+  },
+]);
+
+export const nonSerenityApl = build([
```

Comment:
the way you've structured this, this apl will also apply to serenity actions. you could add a `and(not(inSerenity))` to them all if you want to explicitly exclude serenity.